src/wp-includes/script-loader.php

```diff
@@ -3640,3 +3640,20 @@ function _wp_theme_json_webfonts_handler() {
 	add_action( 'wp_enqueue_scripts', $fn_generate_and_enqueue_styles );
 	add_action( 'admin_init', $fn_generate_and_enqueue_editor_styles );
 }
+
+/**
+ * Loads classic theme styles on classic themes.
+ *
+ * This is needed for backwards compatibility for button blocks specifically.
+ */
+function wp_enqueue_classic_theme_styles() {
+	if ( ! wp_is_block_theme() ) {
+		$suffix = wp_scripts_get_suffix();
+		wp_register_style( 'classic-theme-styles', "/wp-includes/css/dist/block-library/classic$suffix.css", array(), true );
+		wp_enqueue_style( 'classic-theme-styles' );
+	}
+}
+// To load classic theme styles on the frontend.
+add_action( 'wp_enqueue_scripts', 'wp_enqueue_classic_theme_styles' );
+// To load classic theme styles in the the editor.
+add_action( 'admin_enqueue_scripts', 'wp_enqueue_classic_theme_styles' );
```

Comment:
I think these should probably be moved to `wp-includes/default-filters.php` where all other core hooks live.  Probably around https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/default-filters.php#L558

Reply:
I'll handle this in the merge though.

Reply:
Thank you @dream-encode!